modifiers/string.go

```diff
@@ -89,6 +91,35 @@ func SnakeCase(ctx context.Context, t *mold.Transformer, v reflect.Value, param
 	return nil
 }
 
+// NameCase Trims, strips numbers and special characters (except dashes and spaces separating names),
+// converts multiple spaces and dashes to single characters, title cases multiple names.
+// Example: "3493€848Jo-$%£@Ann " -> "Jo-Ann", " ~~ The Dude ~~" -> "The Dude", "**susan**" -> "Susan",
+// " hugh fearnley-whittingstall" -> "Hugh Fearnley-Whittingstall"
+func NameCase(ctx context.Context, t *mold.Transformer, v reflect.Value, param string) error {
+	s, ok := v.Interface().(string)
+	if !ok {
+		return nil
+	}
+	first := onlyOne(strings.ToLower(s), []map[string]string{
+		{`[^\pL-\s']`: ""}, // cut off everything except [ alpha, hyphen, whitespace, apostrophe]
+		{`\s{2,}`: " "},    // trim more than two whitespaces to one
+		{`-{2,}`: "-"},     // trim more than two hyphens to one
+		{`'{2,}`: "'"},     // trim more than two apostrophes to one
+		{`( )*-( )*`: "-"}, // trim enclosing whitespaces around hyphen
+	})
+	v.SetString(strings.Title(regexp.MustCompile(`[\p{L}]([\p{L}|[:space:]|\-|\']*[\p{L}])*`).FindString(first)))
```

Comment:
compiling regexes every invocation is costly, can these be moved to global variables for reuse?

Reply:
I agree. Done.